security-analysis/security-analysis-api/src/main/java/com/powsybl/security/SecurityAnalysisResult.java

```diff
@@ -49,4 +67,16 @@ public LimitViolationsResult getPreContingencyResult() {
     public List<PostContingencyResult> getPostContingencyResults() {
         return postContingencyResults;
     }
+
+    public List<BranchResult> getBranchResultsAsList() {
```

Comment:
In that class too:
* remove the `AsList` suffix
* use `List.copyOf` instead of array list
* provide getters by ID too

But we also need to rename those getters, because it's not clear that they are pre-contingency result.
But I must say it gets a little bit ugly : `getPreContingencyThreeWindingTransformerResults`.

Should we move all those results in a precontingency result objets ? 
If we do it and move the limit violations into it, we need to take care of compatibility (API and JSON format) ...  little bit annoying too.

Any opinion @MioRtia @geofjamg ?

Reply:
Are you thinking about a `PreContingencyResult` object in parallel of `PostContingencyResult`?

Reply:
Yes something like that.